tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -0,0 +1,75 @@
+import httpclient, asynchttpserver, asyncdispatch, asyncfutures
+import net
+
+const postBegin = """
+POST / HTTP/1.1
+Host: 127.0.0.1:64123
+Accept: */*
+Transfer-Encoding:chunked
+Content-Type: application/x-www-form-urlencoded
+
+"""
+
+template genTest(input, expected) =
+  var sanity = false
+  proc handler(request: Request) {.async.} =
+      doAssert(request.body == expected)
+      doAssert(request.headers.hasKey("Transfer-Encoding"))
+      doAssert(not request.headers.hasKey("Content-Length"))
+      sanity = true
+      await request.respond(Http200, "Good")
+
+  let port = Port(64123)
+  let server = newAsyncHttpServer()
+  discard server.serve(port, handler)
```

Comment:
discarding a future seems buggy (eg refs https://github.com/nim-lang/Nim/issues/11912)
/cc @dom96 do you have a suggestion that works here? (neither waitFor nor asyncCheck work here, giving RT errors)

Reply:
@vabresto in the meantime, add a comment:
```nim
# xxx figure out whether there's a better way than discarding a Future here
```
so we can move forward with this PR